### PR TITLE
Update Callback Url Description

### PR DESCRIPTION
**Pull request by atanner27**
Update Callback Url Description to inform the user about the need for localhost to receive the access code

### CODE REVIEW

File: guides/spin/cli/index.md
Comment (dorbin) [inline guides/spin/cli/index.md]: @atanner27 
@jtk54 
"...may involve..." makes this a little vague. Can we predict whether this will be needed, or under what circumstances?

Comment (jtk54) [reply guides/spin/cli/index.md]: This should probably be documented per oauth2 provider.

**Review by dorbin — APPROVED** [review]
Approved pending review from @jtk54

**Review by jtk54 — APPROVED** [review]
LGTM.

**Pull request merged**
